src/test/java/net/snowflake/ingest/streaming/internal/datatypes/IcebergDataTypeTest.java

```diff
@@ -0,0 +1,441 @@
+package net.snowflake.ingest.streaming.internal.datatypes;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.math.BigDecimal;
+import java.sql.ResultSet;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.LocalTime;
+import java.time.OffsetDateTime;
+import java.time.OffsetTime;
+import java.time.ZoneOffset;
+import java.util.Map;
+import java.util.UUID;
+import net.snowflake.ingest.TestUtils;
+import net.snowflake.ingest.streaming.SnowflakeStreamingIngestChannel;
+import net.snowflake.ingest.utils.ErrorCode;
+import net.snowflake.ingest.utils.SFException;
+import org.apache.commons.lang3.StringUtils;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Ignore;
+import org.junit.Test;
+
+@Ignore("This test can be enabled after server side Iceberg EP support is released")
+public class IcebergDataTypeTest extends AbstractDataTypeTest {
+
+  private final ObjectMapper objectMapper = new ObjectMapper();
+
+  @Before
+  public void before() throws Exception {
+    super.before(true);
+  }
+
+  @Test
+  public void testBoolean() throws Exception {
+    testIcebergIngestion("boolean", true, new BooleanProvider());
+    testIcebergIngestion("boolean", false, new BooleanProvider());
+    testIcebergIngestion("boolean", 1, true, new BooleanProvider());
+    testIcebergIngestion("boolean", "false", false, new BooleanProvider());
+
+    SFException ex =
+        Assert.assertThrows(
+            SFException.class,
+            () -> testIcebergIngestion("boolean", new Object(), true, new BooleanProvider()));
+    Assert.assertEquals(ErrorCode.INVALID_FORMAT_ROW.getMessageCode(), ex.getVendorCode());
+  }
+
+  @Test
+  public void testInt() throws Exception {
+    testIcebergIngestion("int", 1, new IntProvider());
+    testIcebergIngestion("int", -.0f, 0, new IntProvider());
+    testIcebergIngestion("int", 0.5f, 1, new IntProvider());
+    testIcebergIngestion("int", "100.4", 100, new IntProvider());
+    testIcebergIngestion("int", new BigDecimal("1000000.09"), 1000000, new IntProvider());
+    testIcebergIngestion("int", Integer.MAX_VALUE, new IntProvider());
+    testIcebergIngestion("int", Integer.MIN_VALUE, new IntProvider());
+
+    SFException ex =
+        Assert.assertThrows(
+            SFException.class,
+            () -> testIcebergIngestion("int", Long.MAX_VALUE, new LongProvider()));
```

Comment:
nit: `(Long)Integer.MAX_VALUE + 1` instead of long.maxvalue?

Reply:
Modified.